compiler/rustc_resolve/messages.ftl

```diff
@@ -114,17 +114,27 @@ resolve_forward_declared_generic_param =
     .label = defaulted generic parameters cannot be forward declared
 
 resolve_generic_params_from_outer_item =
-    can't use generic parameters from outer item
-    .label = use of generic parameter from outer item
+    can't use {$is_self ->
+        [true] `Self`
+        *[false] generic parameters
+    } from outer item
+    .label = use of {$is_self ->
+        [true] `Self`
+        *[false] generic parameter
+    } from outer item
     .refer_to_type_directly = refer to the type directly here instead
     .suggestion = try introducing a local generic parameter here
 
+resolve_generic_params_from_outer_item_const = a `const` creates a new item
```

Comment:
```suggestion
resolve_generic_params_from_outer_item_const = a `const` is a separate item from the body that contains it
```

idk if we should say "body" here, but i think "is separate" is important here. also the verb "creates" seems a bit weird, a `const` *is* an item, it doesn't *create* an item, at least in my mental model.

Reply:
"separate item from that which contains it", maybe?

Reply:
"that which contains it" feels a bit circular 😵

if you don't like "body", maybe just "the item that contains it"